### DISCUSSION TITLE
Shall we rethink Puppetfile format maybe?

### DISCUSSION THREAD

**Discussion by jay7x** [puppetlabs/community#76] [Developer Ecosystem]
I was looking for a puppetfile dependency resolution problem today. What surprised me as a side-effect of this, is that every tool uses its own unique way to deal with Puppetfile. So.. I'd propose to:
1. Rethink the Puppetfile format (as it's executable script actually, not a data file). Maybe it's time to either make it pure data file or to propose another format to replace it.
2. Implement a common library to deal with Puppetfile (including its successor if any). This should be the ONLY recommended way to parse it and it MUST include dependency resolution function.

**Comment by binford2k** [comment]
I am actually working on a data file proposal for this. Tbh, it's kind of silly that we have so many different implementations of more or less the same thing as all or part of these files. I believe they should be standardized with a common schema and parsed & manipulated with a single library.
- fixtures.yml
- Puppetfile
- bolt-project.yaml
- metadata.json
- probably more

Because it's in the _proposal_ state, I cannot promise anything yet.

**Comment by davidsandilands** [comment]
@pmcmaw FYI
